rbac/management/workspace/model.py

```diff
@@ -93,3 +93,28 @@ def _ancestry_queryset(self):
         """,
             [self.id],
         )
+
+    def descendants(self):
+        """Return a list of descendants for a Workspace instance."""
+        descendant_ids = [d.id for d in self._descendant_queryset() if d.id != self.id]
+        descendants = Workspace.objects.filter(id__in=descendant_ids)
+        return descendants
+
+    def _descendant_queryset(self):
```

Comment:
Starting a separate thread to avoid the AI spam heh.

So what's interesting is that as of now we don't actually need the full models in the queries.

I wonder if you could simply do this?

```suggestion
        descendants = Workspace.objects.filter(id__in=self._descendant_queryset())
        return descendants

    def _descendant_queryset(self):
```

This would make it just part of the same SQL query by lazily evaluating the queryset, I think.
It would include your _own_ id, which is not what this does currently, but if we need to exclude that, maybe we could remove that in the CTE:

```
            WITH RECURSIVE descendants AS
              (SELECT id,
                      parent_id
               FROM management_workspace
               WHERE id = %s
               UNION SELECT w.id,
                                w.parent_id
               FROM management_workspace w
               JOIN descendants d ON w.parent_id = d.id)
            SELECT id
            FROM descendants
            WHERE id <> %s
```

(note additional where on the end)